sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/WindowsCredentialApi.java

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation;
+
+import com.sun.jna.Pointer;
+import com.sun.jna.Memory;
+import com.sun.jna.LastErrorException;
+import com.sun.jna.Structure;
+import com.sun.jna.platform.win32.WinBase;
+import com.sun.jna.win32.StdCallLibrary;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class exposes functions from credential manager on Windows platform
+ * via JNA.
+ */
+public interface WindowsCredentialApi extends StdCallLibrary {
+
+    /**
+     * Type of Credential
+     */
+    int CRED_TYPE_GENERIC = 1;
+
+    /**
+     * Credential attributes
+     *
+     * typedef struct _CREDENTIAL_ATTRIBUTE {
+     *   LPTSTR Keyword;
+     *   DWORD  Flags;
+     *   DWORD  ValueSize;
+     *   LPBYTE Value;
+     * } CREDENTIAL_ATTRIBUTE, *PCREDENTIAL_ATTRIBUTE;
+     *
+     */
+    class CREDENTIAL_ATTRIBUTE extends Structure {
```

Comment:
Why are classnames uppercase?

Reply:
These classes map to native C code via Jna.
This is a general best practice when using Native classes to distinguish them.